### PR TITLE
Don't display beta tags for symbols with availability items without introduced version.

### PR DESCRIPTION
**Pull request by sofiaromorales**
Bug/issue #, if applicable: 139245007

## Summary

Don't display beta tags for symbols with availability items without introduced version.

This change makes symbols without introduced version to not be marked as `beta`.

## Dependencies

_Describe any dependencies this PR might have, such as an associated branch in another repository._

## Testing

[Availability.docc.zip](https://github.com/user-attachments/files/17649459/Availability.docc.zip)

Steps:
1. Preview the attached documentation catalog.
2. Assert that the symbols are not marked as beta in the curation.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [X] Added tests
- [X] Ran the `./bin/test` script and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

File: Tests/SwiftDocCTests/Model/SemaToRenderNodeTests.swift [hunk 1]
```diff
@@ -2052,7 +2057,24 @@ Document
...
+                "macOS": PlatformVersion(VersionTriplet(10, 15, 0), beta: true),
+                "watchOS": PlatformVersion(VersionTriplet(6, 0, 0), beta: true),
+                "tvOS": PlatformVersion(VersionTriplet(13, 0, 0), beta: true),
+                "iOS": PlatformVersion(VersionTriplet(13, 0, 0), beta: true)
+            ], referencePath: "/documentation/MyKit")
+            
+            let node = try context.entity(with: reference)
+            let renderNode = try XCTUnwrap(DocumentationNodeConverter(bundle: bundle, context: context).convert(node))
+            
+            // Verify task group link is beta
```
Comment (patshaughnessy) [inline Tests/SwiftDocCTests/Model/SemaToRenderNodeTests.swift:2076]: Comment is misleading? Maybe you copied it from above. You're testing the link is not in beta here.

**Review by patshaughnessy — APPROVED** [review]
Looks good!

File: Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift [hunk 2]
```diff
@@ -219,9 +219,15 @@ public class DocumentationContentRenderer {
         // Verify that if current platforms are in beta, they match the introduced version of the symbol
         for availability in symbolAvailability {
             // If not available on this platform, skip to next platform.
-            guard !availability.isUnconditionallyUnavailable, let introduced = availability.introducedVersion else {
+            guard !availability.isUnconditionallyUnavailable else {
                 continue
             }
+
+            // If the availability item does not have an introduced version,
+            // the symbol can't be marked as beta.
```
Comment (d-ronnqvist) [inline Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift:227]: How about updating this comment to something like this do describe the broader expectation about when a symbol is considered "in beta" and why this code follows those expectations. With the current comment I thought that the early exit could be a bug at first.

```suggestion
            // Symbols are only considered "in beta" if they are in beta for all platforms that they are available for.
            // If the symbol doesn't have an introduced version for one of those platforms, we don't consider it "in beta".
```

**Review by d-ronnqvist — APPROVED** [review]

File: Tests/SwiftDocCTests/Model/SemaToRenderNodeTests.swift (hunk 1 above)
Comment (sofiaromorales) [reply Tests/SwiftDocCTests/Model/SemaToRenderNodeTests.swift]: Yes! thanks for spotting this. Resolved in https://github.com/swiftlang/swift-docc/pull/1086/commits/7571b77740180af7a1ca74fad37a244de92adfff

File: Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift (hunk 2 above)
Comment (sofiaromorales) [reply Sources/SwiftDocC/Model/Rendering/DocumentationContentRenderer.swift]: Resolved [here](https://github.com/swiftlang/swift-docc/pull/1086/commits/c5458a77109ad6c8f263c4268bdb5586701f5a9e)

**Review by franklinsch — APPROVED** [review]

**Comment by sofiaromorales** [comment]
@swift-ci please test

**Pull request merged**